### ISSUE TITLE
Cannot bind to both IPv4 and IPv6 address

### ISSUE THREAD

**Issue by donpatrice** [badaix/snapcast#518]
As far as I understand the comments in `/etc/snapserver.conf` it should be possible to specify both, an IPv4 and an IPv6 address like this:

```
$ cat /etc/snapserver.conf | grep -B9 bind_to_address
[http]
# enable HTTP Json RPC (HTTP POST and websockets)
#enabled = true

# address to listen on, can be specified multiple times
# use "0.0.0.0" to bind to any IPv4 address or :: to bind to any IPv6 address
# or "127.0.0.1" or "::1" to bind to localhost IPv4 or IPv6, respectively
# use the address of a specific network interface to just listen to and accept
# connections from that interface
bind_to_address = ::
bind_to_address = 0.0.0.0
--
[tcp]
# enable TCP Json RPC
#enabled = true

# address to listen on, can be specified multiple times
# use "0.0.0.0" to bind to any IPv4 address or :: to bind to any IPv6 address
# or "127.0.0.1" or "::1" to bind to localhost IPv4 or IPv6, respectively
# use the address of a specific network interface to just listen to and accept
# connections from that interface
bind_to_address = ::
bind_to_address = 0.0.0.0
--
# Stream settings #############################################################
#
[stream]
# address to listen on, can be specified multiple times
# use "0.0.0.0" to bind to any IPv4 address or :: to bind to any IPv6 address
# or "127.0.0.1" or "::1" to bind to localhost IPv4 or IPv6, respectively
# use the address of a specific network interface to just listen to and accept
# connections from that interface
bind_to_address = ::
bind_to_address = 0.0.0.0
```
But on the other hand I get this:
```
$ sudo netstat -tulpen | grep snap
tcp6       0      0 :::1704                 :::*                    LISTEN      110        17706      1115/snapserver     
tcp6       0      0 :::1705                 :::*                    LISTEN      110        17696      1115/snapserver     
tcp6       0      0 :::1780                 :::*                    LISTEN      110        17698      1115/snapserver  
```
Interchanging the order of the IPv4 and IPv6 lines I get:
```
$ sudo netstat -tulpen | grep snap
tcp        0      0 0.0.0.0:1704            0.0.0.0:*               LISTEN      110        19029      1255/snapserver     
tcp        0      0 0.0.0.0:1705            0.0.0.0:*               LISTEN      110        19019      1255/snapserver     
tcp        0      0 0.0.0.0:1780            0.0.0.0:*               LISTEN      110        19021      1255/snapserver
```
So it seems `bind_to_address` is only interpreted once. Is this expected?

**Comment by badaix** [comment]
I tried it on my Ubuntu machine, and it seems that binding to IPv4 only binds to IPv4, while IPv6 binds to both, IPv4 and IPv6 (there exists a socket option `IPV6_V6ONLY` that would cause IPv6 to IPv6 only)
This means:
1. `bind_to_address = 0.0.0.0` will accept IPv4 connections only
2. `bind_to_address = ::` will accept IPv6 and IPv4 as mapped IPv6 connections
3. `bind_to_address = 0.0.0.0` and `bind_to_address = ::` will bind to IPv4 and fail to bind to IPv6 (because it implicitly wants to bind to IPv4 too):  
```
2019-12-26 23-23-57 [Info] Creating stream acceptor for address: 0.0.0.0, port: 1704
2019-12-26 23-23-57 [Info] Creating stream acceptor for address: ::, port: 1704
2019-12-26 23-23-57 [Err] error creating TCP acceptor: bind: Address already in use, code: system:98
```
4. `bind_to_address = ::` and `bind_to_address = 0.0.0.0`  will accept IPv6 and IPv4 as mapped IPv6 connections and will fail to bind to IPv4 only (address is already in use by the mapped IPv4 part of the IPv6 connection)  
```
2019-12-26 23-26-32 [Info] Creating stream acceptor for address: ::, port: 1704
2019-12-26 23-26-32 [Info] Creating stream acceptor for address: 0.0.0.0, port: 1704
2019-12-26 23-26-32 [Err] error creating TCP acceptor: bind: Address already in use, code: system:98
```
Long story short: The comment in the conf file should be refined and if you want to listen to IPv4 and IPv6 connections, you should bind to the IPv6 address

**Comment by donpatrice** [comment]
Great, thanks a lot. Only binding to the IPv6 address, only shows:
```
$ sudo netstat -tulpen | grep snap
tcp        0      0 0.0.0.0:1704            0.0.0.0:*               LISTEN      110        19029      1255/snapserver     
tcp        0      0 0.0.0.0:1705            0.0.0.0:*               LISTEN      110        19019      1255/snapserver     
tcp        0      0 0.0.0.0:1780            0.0.0.0:*               LISTEN      110        19021      1255/snapserver
```
but the IPv4 address is mapped to the IPv6 address anyway.
